test/ttmlir/Dialect/TTNN/sharding_matmul_32.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="enable-optimizer=true sharding-pass-enabled=true max-legal-layouts=32" %s | FileCheck %s
+#any_device_tile = #tt.operand_constraint<dram|l1|tile|any_device_tile>
+module attributes {} {
+  func.func @forward(%arg0: tensor<64x128xbf16>, %arg1: tensor<128x96xbf16>, %arg2: tensor<96x64xbf16>) -> tensor<64x64xbf16> {
+    // CHECK: #[[L1_:.*]] = #tt.memory_space<l1>
+    // CHECK: #[[LAYOUT_7:.*]] = #tt.layout<(d0, d1) -> (d0, d1), undef, <2x3>, memref<1x1x!tt.tile<32x32, bf16>, #l1_>, block_sharded>
+    // CHECK: #[[LAYOUT_8:.*]] = #tt.layout<(d0, d1) -> (d0, d1), undef, <8x8>, memref<1x1x!tt.tile<32x32, bf16>, #dram>, interleaved>
+    %0 = tensor.empty() : tensor<64x96xbf16>
+    // CHECK: %[[C:.*]] = "ttnn.matmul"[[C:.*]] -> tensor<64x96xbf16, #[[LAYOUT_7]]>
+    %1 = "ttir.matmul"(%arg0, %arg1, %0) <{operand_constraints = [#any_device_tile, #any_device_tile, #any_device_tile]}> : (tensor<64x128xbf16>, tensor<128x96xbf16>, tensor<64x96xbf16>) -> tensor<64x96xbf16>
+    %2 = tensor.empty() : tensor<64x64xbf16>
+    // CHECK: %[[C:.*]] = "ttnn.matmul"[[C:.*]] -> tensor<64x64xbf16, #[[LAYOUT_8]]>
```

Comment:
You can remove this check since we don't require it to be dram interleaved in order for the override to be working properly. It is completely fine to check the first op only.